poetry/packages/locker.py

```diff
@@ -176,20 +176,26 @@ def locked_repository(
                         package.marker = parse_marker(split_dep[1].strip())
 
             for dep_name, constraint in info.get("dependencies", {}).items():
+
+                root_dir = self._lock.path.parent
+                if package.source_type == "directory":
+                    # root dir should be the source of the package relative to the lock path
+                    root_dir = Path(
+                        os.path.relpath(
```

Comment:
Couple of points;
1. Move to using pathlib if possible.
2. We cannot always be sure a "directory" package is relative as it could also be an absolute path.